agent/test/unit/com/thoughtworks/go/agent/AgentProcessParentImplTest.java

```diff
@@ -278,14 +275,10 @@ private File randomFile(final String pathname) {
     }
 
     private void createRandomFile(File agentJar, final String data) {
-        FileOutputStream output = null;
-        try {
-            output = new FileOutputStream(agentJar);
+        try (FileOutputStream output = new FileOutputStream(agentJar)) {
             IOUtils.write(data, output);
         } catch (IOException e) {
-            throw new RuntimeException(e);
-        } finally {
-            IOUtils.closeQuietly(output);
+            e.printStackTrace();
```

Comment:
I'm not reviewing this. Someone else will. But, I have a question about this line. Why? Why does it not throw any more? Shouldn't matter that much in this test, but still.

The only other part that seems a little risky here is the change to applicationContext, which needs to be verified (to make sure the timers are triggering as expected, and there is no difference between the old and the new). For instance, the behavior when one of the scheduled task runs takes more than the interval time - will it get queued vs. will it get run anyway.